src/app/seats-chart/seats-chart.component.ts

```diff
@@ -1,20 +1,49 @@
 import { Component, inject } from '@angular/core';
+import { combineLatest } from 'rxjs';
+import { I18nService } from '@ui5/webcomponents-ngx/i18n';
 
-import { ALPHABETS } from 'src/assets/constant-querries';
-import { CURRENT_TRIP } from 'src/assets/mock-data/mock-trips';
-import { DEPARTURE_AIRCRAFT_STATUS } from "src/assets/mock-data/mock-aircraft-status"
+import { AppService } from '../services/services';
+import { ALPHABETS } from '../constants/constants';
 import { LegendItem } from '../interfaces/legend-item';
-import { I18nService } from '@ui5/webcomponents-ngx/i18n';
 
 @Component({
     selector: 'app-seats-chart',
     templateUrl: './seats-chart.component.html',
     styleUrls: ['./seats-chart.component.scss']
 })
 export class SeatsChartComponent {
-    constructor() { }
+    isDataAvailable = false;
+
+    i18nService = inject(I18nService);
+
+    departureAircraftStatus: any;
+    rows: any;
+    columns: any;
+    columnLabelIndex: any;
+    availableSeats: any;
+    yourSeats: any;
+    chars = ALPHABETS;
+    legendItems: LegendItem[] = [
+        { icon: "sys-enter-2", color: "legend-item__icon--informative", text: "YOUR_SEATS" },
+        { icon: "circle-task-2", color: "", text: "TAKEN" },
+        { icon: "circle-task", color: "legend-item__icon--message", text: "AVAILABLE" }
+    ];
 
-    ngOnInit() { }
+    constructor(private appService: AppService) { }
+
+    ngOnInit() {
+        combineLatest([this.appService.getCurrentTrip(), this.appService.getDepartureAircraftStatus()])
+            .subscribe(([currentTrip, departureAircraftStatus]) => {
+                this.yourSeats = currentTrip.seatsSelected;
+                this.departureAircraftStatus = departureAircraftStatus;
+                this.rows = this.departureAircraftStatus.rows;
+                this.columns = this.departureAircraftStatus.columns;
+                this.columnLabelIndex = this.sumPrefix(this.columns);
+                this.availableSeats = this.departureAircraftStatus.availableSeats;
+
+                this.isDataAvailable = true;
+            });
```

Comment:
If you do not have any combination of the results from both services, then `zip` is fine I guess